Userland/Libraries/LibFileSystemAccessClient/Client.h

```diff
@@ -46,10 +49,14 @@ class Client final
     virtual void handle_prompt_end(i32 request_id, i32 error, Optional<IPC::File> const& fd, Optional<DeprecatedString> const& chosen_file) override;
 
     int get_new_id();
-    DeprecatedResult handle_promise(int);
+    template<typename AnyResult>
+    AnyResult handle_promise(int);
+
+    template<typename T>
+    using PromiseType = RefPtr<Core::Promise<T>>;
 
     struct PromiseAndWindow {
-        RefPtr<Core::Promise<DeprecatedResult>> promise {};
+        Variant<Empty, PromiseType<DeprecatedResult>, PromiseType<Result>> promise {};
```

Comment:
Where do we even use the `Empty` type?

Reply:
On the same line :^)

The compiler doesn't let me keep the `{}` without `Empty`.
I removed it, but that's weird though

Reply:
It works just fine without the `Empty` type if we remove the initialization with `{}` and instead make sure that we always initialize it with either of both types whenever we actually instantiate a `PromiseAndWindow`.